### PR TITLE
batching delegations and undelegations

### PR DESCRIPTION
**Pull request by sampocs**
## Context
For the v24 upgrade, we're shipping the batching changes, but we can't ship the stTIA migration changes until Celestia adds ICA. This PR is the equivalent to #1223 but with all the stTIA migration changes reverted.

### CODE REVIEW

File: x/stakeibc/keeper/icacallbacks_claim.go [hunk 1]
```diff
@@ -41,7 +40,7 @@ func (k Keeper) ClaimCallback(ctx sdk.Context, packet channeltypes.Packet, ackRe
 	// Fetch callback args
 	claimCallback, err := k.UnmarshalClaimCallbackArgs(ctx, args)
 	if err != nil {
-		return errorsmod.Wrapf(types.ErrUnmarshalFailure, fmt.Sprintf("Unable to unmarshal claim callback args: %s", err.Error()))
+		return err
```
Comment (riley-stride) [inline x/stakeibc/keeper/icacallbacks_claim.go:43]: why not log?

File: x/stakeibc/keeper/icacallbacks_redemption.go [hunk 2]
```diff
@@ -43,7 +42,7 @@ func (k Keeper) RedemptionCallback(ctx sdk.Context, packet channeltypes.Packet,
 	// Fetch callback args
 	redemptionCallback, err := k.UnmarshalRedemptionCallbackArgs(ctx, args)
 	if err != nil {
-		return errorsmod.Wrapf(types.ErrUnmarshalFailure, fmt.Sprintf("Unable to unmarshal redemption callback args: %s", err.Error()))
+		return err
```
Comment (riley-stride) [inline x/stakeibc/keeper/icacallbacks_redemption.go:45]: why not log?

File: x/stakeibc/keeper/icacallbacks_reinvest.go [hunk 3]
```diff
@@ -53,7 +52,7 @@ func (k Keeper) ReinvestCallback(ctx sdk.Context, packet channeltypes.Packet, ac
 	// Fetch callback args
 	reinvestCallback, err := k.UnmarshalReinvestCallbackArgs(ctx, args)
 	if err != nil {
-		return errorsmod.Wrapf(types.ErrUnmarshalFailure, fmt.Sprintf("Unable to unmarshal reinvest callback args: %s", err.Error()))
+		return err
```
Comment (riley-stride) [inline x/stakeibc/keeper/icacallbacks_reinvest.go:55]: why not log?

File: x/stakeibc/keeper/epoch_tracker.go [hunk 4]
```diff
@@ -100,41 +100,32 @@ func (k Keeper) GetStrideEpochElapsedShare(ctx sdk.Context) (sdk.Dec, error) {
...
-		k.Logger(ctx).Error(errMsg)
-		return sdk.ZeroDec(), errorsmod.Wrapf(types.ErrIntCast, errMsg)
+		return sdk.ZeroDec(), errorsmod.Wrap(err, "unable to convert epoch duration to int64")
 	}
 	epochEndTime, err := cast.ToInt64E(epochTracker.NextEpochStartTime)
 	if err != nil {
-		errMsg := fmt.Sprintf("unable to convert next epoch start time to int64, err: %s", err.Error())
-		k.Logger(ctx).Error(errMsg)
-		return sdk.ZeroDec(), errorsmod.Wrapf(types.ErrIntCast, errMsg)
+		return sdk.ZeroDec(), errorsmod.Wrapf(err, "unable to convert next epoch start time to int64")
```
Comment (riley-stride) [inline x/stakeibc/keeper/epoch_tracker.go:113]: nit: Wrap

File: x/icacallbacks/migrations/v2/convert.go [hunk 5]
```diff
@@ -74,14 +74,14 @@ func convertCallbackData(oldCallbackData icacallbacktypes.CallbackData) (icacall
 		// Deserialize the callback args with the old DelegateCallback type
 		oldDelegateCallback := oldstakeibctypes.DelegateCallback{}
 		if err := proto.Unmarshal(oldCallbackData.CallbackArgs, &oldDelegateCallback); err != nil {
-			return icacallbacktypes.CallbackData{}, errorsmod.Wrapf(stakeibctypes.ErrUnmarshalFailure, err.Error())
+			return icacallbacktypes.CallbackData{}, errorsmod.Wrapf(err, "failed to unmarshal")
```
Comment (riley-stride) [inline x/icacallbacks/migrations/v2/convert.go:77]: Nit: this and all the below should be Wrap

File: x/icaoracle/types/genesis.go [hunk 6]
```diff
@@ -17,7 +17,7 @@ func DefaultGenesis() *GenesisState {
 // failure.
 func (gs GenesisState) Validate() error {
 	if err := gs.Params.Validate(); err != nil {
-		return errorsmod.Wrapf(ErrInvalidGenesisState, err.Error())
+		return errorsmod.Wrapf(err, "invalid genesis state")
```
Comment (riley-stride) [inline x/icaoracle/types/genesis.go:20]: nit: Wrap

**Review by riley-stride — APPROVED** [review]
Lgtm once nit comments are addressed

File: x/stakeibc/keeper/epoch_tracker.go (hunk 4 above)
Comment (sampocs) [reply x/stakeibc/keeper/epoch_tracker.go]: done

File: x/icacallbacks/migrations/v2/convert.go (hunk 5 above)
Comment (sampocs) [reply x/icacallbacks/migrations/v2/convert.go]: done

File: x/interchainquery/keeper/keeper.go [hunk 7]
```diff
@@ -96,7 +96,7 @@ func (k *Keeper) RetryICQRequest(ctx sdk.Context, query types.Query) error {
 
 	// Submit a new query (with a new ID)
 	if err := k.SubmitICQRequest(ctx, query, true); err != nil {
-		return errorsmod.Wrapf(err, types.ErrFailedToRetryQuery.Error())
+		return errorsmod.Wrapf(err, "failed to retry query")
```
[the remark this answers is not in the document]
Comment (sampocs) [reply x/interchainquery/keeper/keeper.go]: done

File: x/icaoracle/types/genesis.go (hunk 6 above)
Comment (sampocs) [reply x/icaoracle/types/genesis.go]: done

**Pull request merged**